src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -167,16 +289,59 @@ begin
   { rw [← v.equiv_fun.symm_apply_apply y, v.equiv_fun_symm_apply] }
 end
 
-@[simp] lemma basis.coe_isometry_euclidean_of_orthonormal
+
+@[simp] lemma _root_.basis.coe_isometry_euclidean_of_orthonormal
   (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
   (v.isometry_euclidean_of_orthonormal hv : E → euclidean_space 𝕜 ι) = v.equiv_fun :=
 rfl
 
-@[simp] lemma basis.coe_isometry_euclidean_of_orthonormal_symm
+@[simp] lemma _root_.basis.coe_isometry_euclidean_of_orthonormal_symm
   (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
   ((v.isometry_euclidean_of_orthonormal hv).symm : euclidean_space 𝕜 ι → E) = v.equiv_fun.symm :=
 rfl
 
+
+/-- An orthonormal set that spans is an orthonormal basis -/
+protected def mk (hon : orthonormal 𝕜 v) (hsp: submodule.span 𝕜 (set.range v) = ⊤):
+  orthonormal_basis ι 𝕜 E :=
+    orthonormal_basis.of_repr $
+    basis.isometry_euclidean_of_orthonormal (basis.mk (orthonormal.linear_independent hon) hsp)
```

Comment:
Instead of doing this via `basis.isometry_euclidean_of_orthonormal`, I think you should actually delete `basis.isometry_euclidean_of_orthonormal` and move the contents here!  Since the two are functionally equivalent.  This will break a fair bit elsewhere in the library, but nothing should be hard to fix ... let me know if you need help.

In fact, it will probably work out more elegantly if you first define `basis.to_orthonormal_basis` as the direct successor to `basis.isometry_euclidean_of_orthonormal`, and then make this construction a consequence of `basis.to_orthonormal_basis` using `(basis.mk (orthonormal.linear_independent hon) hsp)`.